javaslang/src/main/java/javaslang/collection/Maps.java

```diff
@@ -238,6 +240,20 @@
         return replace(map, currentElement, newElement);
     }
 
+    static <K, V, M extends Map<K, V>> M replaceValue(M map, K key, V value) {
+        return map.containsKey(key) ? put(map, Tuple(key, value)) :  map;
```

Comment:
> I would rather use the first version and avoid the cast to M. Is that OK?

Please cast `(M) map.put(key, value)`. We save an additional Tuple instance which scales for many calls. I see our collections as low-level, they should be as performant as possible. A cast is fast.